motionplan/motionPlanner.go

```diff
@@ -381,3 +382,81 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan []map[string][]frame.Input,
+	worldState *frame.WorldState,
```

Comment:
Is world state allowed to be nil? If not, should this function validate that it is not nil?

Reply:
WrldSt is allowed to be nil.
```
	t.Run("base case - validate plan without obstacles", func(t *testing.T) {
		err := CheckPlan(ackermanFrame, steps, nil, fs, startPose, errorState, logger)
		test.That(t, err, test.ShouldBeNil)
	})
```
found in `tpSpaceRRT_test.go`